contrib/Shopify/sarama/option.go

```diff
@@ -7,15 +7,24 @@ package sarama
 
 import (
 	"math"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/globalconfig"
 )
 
 type config struct {
-	serviceName   string
-	analyticsRate float64
+	consumerServiceName string
+	producerServiceName string
+	analyticsRate       float64
 }
 
 func defaults(cfg *config) {
-	cfg.serviceName = "kafka"
+	cfg.consumerServiceName = "kafka"
+	cfg.producerServiceName = globalconfig.ServiceName()
```

Comment:
I see, so we can't just have `serviceName` because this global config is used for wrapping both the consumer and producer clients?

I would have thought we'd have two separate `config` objects, one for the consumer and one for the producer.

Reply:
Also shouldn't the consumer use the globally configured service name? Since the consumer is likely the actual service running, whereas the producer is just a client?

Reply:
Yes, they use the same config, so we have to differentiate.

The decision to do the producer and not the consumer was made by @brettlangdon. I can see it either way, and I could also imagine considering both clients to an external service. I'm happy with whatever we decide, but for the metrics tagging, we tagged the producer.